src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -1,410 +0,0 @@
-    /// This listener does not poll very often, and thus takes time to enable and disable tracing.  This is by design to ensure that the listener does
-    /// not starve other threads on the system.
-    /// NOTE: If COMPlus_EnableEventPipe != 4 then this listener is not created and does not add any overhead to the process.
-    /// </summary>
-    internal sealed class EventPipeController
-    {
-        // Miscellaneous constants.
-        private const string DefaultAppName = "app";
-        private const string NetPerfFileExtension = ".netperf";
-        private const string ConfigFileSuffix = ".eventpipeconfig";
-        private const int EnabledPollingIntervalMilliseconds = 1000; // 1 second
-        private const int DisabledPollingIntervalMilliseconds = 20000; // 20 seconds
-        private const uint DefaultCircularBufferMB = 1024; // 1 GB
-        private const char ConfigEntryDelimiter = '=';
-        private const char ProviderConfigDelimiter = ',';
-        private const char ConfigComponentDelimiter = ':';
-
-        // Config file keys.
-        private const string ConfigKey_Providers = "Providers";
-        private const string ConfigKey_CircularMB = "CircularMB";
-        private const string ConfigKey_OutputPath = "OutputPath";
-        private const string ConfigKey_ProcessID = "ProcessID";
-
-        // The default set of providers/keywords/levels.  Used if an alternative configuration is not specified.
-        private static EventPipeProviderConfiguration[] DefaultProviderConfiguration => new EventPipeProviderConfiguration[]
-        {
-            new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntime", 0x4c14fccbd, 5, null),
-            new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntimePrivate", 0x4002000b, 5, null),
-            new EventPipeProviderConfiguration("Microsoft-DotNETCore-SampleProfiler", 0x0, 5, null),
-        };
-
-        // Singleton controller instance.
-        private static EventPipeController? s_controllerInstance;
-
-        // Controller object state.
-        private Timer m_timer;
-        private string m_configFilePath;
-        private bool m_configFileExists;
-
-        // Initialization flag used to avoid initializing FrameworkEventSource on the startup path.
-        internal static bool Initializing { get; private set; }
-
-        internal static void Initialize()
-        {
-            // Don't allow failures to propagate upstream.  Ensure program correctness without tracing.
-            try
-            {
-                Initializing = true;
-
-                if (s_controllerInstance == null)
-                {
-                    int enabled = Config_EnableEventPipe;
-                    if (enabled > 0)
-                    {
-                        // Enable tracing immediately.
-                        // It will be disabled automatically on shutdown.
-                        EventPipe.Enable(BuildConfigFromEnvironment());
```

Comment:
Is there any alternate code path that enables event pipe based on the environment variable? If not I recommend you leave this functionality intact for now.

Reply:
Leave the non-timer/file based functionality to control EventPipe via COMPLus_*?